src/njoy21/input/THERMR/Card2/test/Card2.test.cpp

```diff
@@ -0,0 +1,95 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include "njoy21.hpp"
+
+using namespace njoy::njoy21::input;
+
+SCENARIO( "THERMR Card2 inputs",
+  "[THERMR], [Card2] "){
+  GIVEN( "a nonzero Card1 nendf input" ){
+    Argument< THERMR::Card1::Nendf > nendf; nendf.value = 3;
+    WHEN( "input is valid, all values provided" ){
+      iRecordStream<char> is( std::istringstream("1 125 8 2 2 0 0 2 222 1") );
+      THEN( "the extracted values are correct" ){
+        THERMR::Card2 card2( is, nendf );
+        REQUIRE( card2.matde.value == 1 );
+        REQUIRE( card2.matdp.value == 125 );
+        REQUIRE( card2.nbin.value == 8 );
+        REQUIRE( card2.ntemp.value == 2 );
+        REQUIRE( card2.iin.value == 2 );
+        REQUIRE( card2.icoh.value == 0 );
+        REQUIRE( card2.iform.value == 0 );
+        REQUIRE( card2.natom.value == 2 );
+        REQUIRE( card2.mtref.value == 222 );
+        REQUIRE( card2.iprint.value == 1 );
+      } // THEN 
+    } // WHEN
+
+    WHEN( "input is valid, optional inputs not included" ){
+      iRecordStream<char> is( std::istringstream("1 125 8 2 2 0 0 2 222 /") );
+      THEN( "default values used" ){
+        THERMR::Card2 card2( is, nendf );
+	REQUIRE( card2.iprint.value == 0 );
+      } // THEN
+    } // WHEN
+    
+    WHEN( "matde conflicts with card1 nendf value" ){
+      iRecordStream<char> is( std::istringstream("0 125 8 2 2 0 0 2 222 1") );
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( THERMR::Card2( is, nendf ) );
+      } // THEN
+    } // WHEN
+ 
+    WHEN( "an input tape is out of range" ){
```

Comment:
What do you mean? I wrote each of them in turn so that for the iRecordStream named is_n_, the _n_th input is out of its defined range. For example, is9 has an input mtref of 220, which is out of its defined range of 221-250.